library/pod-security-policy/selinux/template.yaml

```diff
@@ -56,89 +56,147 @@ spec:
                     description: "An SELinux user."
   targets:
     - target: admission.k8s.gatekeeper.sh
-      rego: |
-        package k8spspselinux
+      code:
+      - engine: K8sNativeValidation
+        source:
+          variables:
+          - name: usingAllowedSELinuxOptions
+            expression: |
+              has(variables.anyObject.spec.securityContext) && has(variables.anyObject.spec.securityContext.seLinuxOptions) ? 
+                (has(variables.params.allowedSELinuxOptions) ? 
+                (
+                  size(variables.params.allowedSELinuxOptions) > 0 && variables.params.allowedSELinuxOptions.all(opts,
```

Comment:
I don't think .all() is correct here, the way the rego reads, at least one provided `allowedSELinuxOptions` must be satisfied.

As always, verify with testing.